crypto/src/main/scala/coop/rchain/crypto/util/CertificateHelper.scala

```diff
@@ -110,14 +120,68 @@ object CertificateHelper {
     val cert = new X509CertImpl(info)
     cert.sign(privateKey, algorythm)
 
-    // Update the algorith, and resign.
+    // Update the algorithm, and resign.
     val algorithmId2 = cert.get(X509CertImpl.SIG_ALG).asInstanceOf[AlgorithmId]
     info.set(CertificateAlgorithmId.NAME + "." + CertificateAlgorithmId.ALGORITHM, algorithmId2)
     val cert2 = new X509CertImpl(info)
     cert2.sign(privateKey, algorythm)
     cert2
   }
 
+  def encodeSignatureRStoDER(signatureRS: Array[Byte]): Either[Throwable, Array[Byte]] = {
+    def toASN1Int(bytes: Array[Byte]) = new ASN1Integer(BigIntegers.fromUnsignedByteArray(bytes))
+
+    def convert = {
+      val (r, s) = signatureRS.take(64).splitAt(32)
+      val bos    = new ByteArrayOutputStream(72)
+      val seq    = new DERSequenceGenerator(bos)
+      try {
+        seq.addObject(toASN1Int(r))
+        seq.addObject(toASN1Int(s))
+        // Close to write the sequence
+        seq.close
```

Comment:
@marcin-rzeznicki it seems that closing a sequence is only for writing the data and not for release any resource so it's not necessary in the finally block.

Reply:
Makes sense, :+1: